api/server.js

```diff
@@ -0,0 +1,140 @@
+require('dotenv').config();
+
+const express = require('express');
+const helmet = require('helmet');
+const knex = require('knex');
+const bcrypt = require('bcryptjs'); // added
+const jwt = require('jsonwebtoken');
+const knexConfig = require('../knexfile.js');
+
+const server = express();
+
+const db = knex(knexConfig.development);
+
+server.use(helmet());
+server.use(express.json());
+
+server.get('/', (req, res) => {
+  res.send('hopefully everyone is up in here, up in here.');
+});
+
+server.post('/register', (req, res) => {
+  const userInfo = req.body;
+  // userinfo consists of username, name, and password
+  const hash = bcrypt.hashSync(userInfo.password, 12);
+
+  userInfo.password = hash;
+
+  db('users')
+    .insert(userInfo)
+    .then(ids => {
+      res.status(201).json(ids);
+    })
+    .catch(err => res.status(500).json(err));
+});
+
+///////=============== Middleware ===============///////
+// might want to clean up the routes and put these functions into the dbHelpers in the future
+
+function makeToken(user) {
```

Comment:
Pretty sure you could `function makeToken({username, name}) {` to destructure then on line 42 & 43 you'd just have to do `username: username` and `name: name`. (Those could turn into just `username, name` too).